be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -274,4 +274,32 @@ void ScanOperatorFactory::close(RuntimeState* state) {
     OperatorFactory::close(state);
 }
 
+// ====================
+
+pipeline::OpFactories decompose_scan_node_to_pipeline(std::shared_ptr<ScanOperatorFactory> scan_operator,
```

Comment:
very excellent refactoring

Reply:
Can not be done without your help! Thanks for your constant pushing!